WIP: Examples using Promises/RxJs for for Deploy Smart Contract and Change Runtime Code

ltfschoen opened the pull request:
* See FIXME's in the code where errors or unexpected results are occurring
* Refer to PR #292

jacogr commented on docs/examples/promise/09_change_runtime_code/index.js:
```diff
@@ -0,0 +1,97 @@
+  const referendumPublicId = await api.tx.democracy.startReferendum(proposalSubmitted, 0);
+  console.log(`Public Proposed Referendum with ID started: ${referendumPublicId}`);
+
+  const referendumCount = await api.query.chain.referendumCount();
```
Instead of having all these seperate, we should rather do a 

```
const [a, b, c] = Promise.all([
  api...,
  api....,
  ...
]);

console.log(`a=${a}, b=${b} ...`);
```

jacogr commented on docs/examples/rx/09_change_runtime_code/index.js:
```diff
@@ -0,0 +1,135 @@
+    .pipe(
+      // `concatMap` subscribes to observables in order
+      concatMap((api) =>
+        combineLatest([
```
This is an unworkable example, looks like callback hell :)

ltfschoen replied:
@amaurymartiny I'm getting the following error and not sure how to overcome the callback hell-like beast that i've created. Any ideas? I'm also getting the following error:
```
/Users/me/code/src/polkadot-js/api/docs/examples/rx/09_change_runtime_code/node_modules/rxjs/internal/util/hostReportError.js:4
    setTimeout(function () { throw err; });
                             ^

TypeError: api.tx.consensus.setCode(...).pipe is not a function
    at MergeMapSubscriber.api.pipe.concatMap [as project] (/Users/me/code/src/polkadot-js/api/docs/examples/rx/09_change_runtime_code/index.js:43:14)
    at MergeMapSubscriber._tryNext (/Users/me/code/src/polkadot-js/api/docs/examples/rx/09_change_runtime_code/node_modules/rxjs/internal/operators/mergeMap.js:69:27)
    at MergeMapSubscriber._next (/Users/me/code/src/polkadot-js/api/docs/examples/rx/09_change_runtime_code/node_modules/rxjs/internal/operators/mergeMap.js:59:18)
    at MergeMapSubscriber.Subscriber.next (/Users/me/code/src/polkadot-js/api/docs/examples/rx/09_change_runtime_code/node_modules/rxjs/internal/Subscriber.js:67:18)
    at /Users/me/code/src/polkadot-js/api/docs/examples/rx/09_change_runtime_code/node_modules/rxjs/internal/util/subscribeToPromise.js:7:24
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

jacogr commented on docs/examples/promise/08_deploy_smart_contract/index.js:
```diff
@@ -0,0 +1,32 @@
+  console.log(`Contract deployed for Alice: ${chainDeploymentHash}`);
+
+  // Retrieve Smart Contract
+  // FIXME: Error: Method: cannot decode value ""0x2001000310deadbeef"".
```
Cannot go in, not working I believe.

jacogr replied:
Addressed in https://github.com/polkadot-js/api/pull/334

ltfschoen commented:
@jacogr @amaurymartiny I've pushed a commit that uses `Promise.all` in the promises examples to address Jacos review comment. I've also removed use of async in the examples (as request by Amaury in https://github.com/polkadot-js/api/pull/292). I've also updated all the dependencies used in the examples.

However I'm now getting the following error in both the promises and rx examples when `api.tx.contract.create` or `api.tx.democracy.propose` are called. However I **do not** get an error if I run `api.tx.consensus.setCode` or `api.tx.democracy.startReferendum`:

```
ExtError: Unable to determine type from '::Type'
    at assert (/Users/me/code/src/polkadot-js/api/docs/examples/rx/08_deploy_smart_contract/node_modules/@polkadot/util/assert.js:44:9)
    at getTypeClass (/Users/me/code/src/polkadot-js/api/docs/examples/rx/08_deploy_smart_contract/node_modules/@polkadot/types/codec/createType.js:148:20)
    at Method.filterOrigin.reduce (/Users/me/code/src/polkadot-js/api/docs/examples/rx/08_deploy_smart_contract/node_modules/@polkadot/types/Method.js:136:49)
    at Array.reduce (<anonymous>)
    at Function.getArgsDef (/Users/me/code/src/polkadot-js/api/docs/examples/rx/08_deploy_smart_contract/node_modules/@polkadot/types/Method.js:132:38)
    at Function.decodeMethod (/Users/me/code/src/polkadot-js/api/docs/examples/rx/08_deploy_smart_contract/node_modules/@polkadot/types/Method.js:92:30)
    at new Method (/Users/me/code/src/polkadot-js/api/docs/examples/rx/08_deploy_smart_contract/node_modules/@polkadot/types/Method.js:45:28)
    at extrinsicFn (/Users/me/code/src/polkadot-js/api/docs/examples/rx/08_deploy_smart_contract/node_modules/@polkadot/extrinsics/utils/createUnchecked.js:32:15)
    at Object.decorated [as create] (/Users/me/code/src/polkadot-js/api/docs/examples/rx/08_deploy_smart_contract/node_modules/@polkadot/api/rx/index.js:207:76)
    at SwitchMapSubscriber.api.pipe.switchMap [as project] (/Users/me/code/src/polkadot-js/api/docs/examples/rx/08_deploy_smart_contract/index.js:27:25)
```

jacogr commented:
As per the previous PR comments, https://github.com/polkadot-js/api/pull/292#issuecomment-433722492

jacogr commented:
Closing as stale. Re-open as required.
